client/rest-high-level/src/main/java/org/elasticsearch/client/Request.java

```diff
@@ -0,0 +1,104 @@
+/*
+ * Licensed to Elasticsearch under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.elasticsearch.client;
+
+import org.apache.http.HttpEntity;
+import org.elasticsearch.action.get.GetRequest;
+import org.elasticsearch.common.Strings;
+import org.elasticsearch.common.lucene.uid.Versions;
+import org.elasticsearch.index.VersionType;
+import org.elasticsearch.search.fetch.subphase.FetchSourceContext;
+
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.Locale;
+import java.util.Map;
+import java.util.StringJoiner;
+
+final class Request {
+
+    final String method;
+    final String endpoint;
+    final Map<String, String> params;
+    final HttpEntity entity;
+
+    private Request(String method, String endpoint, Map<String, String> params, HttpEntity entity) {
+        this.method = method;
+        this.endpoint = endpoint;
+        this.params = params;
+        this.entity = entity;
+    }
+
+    @Override
+    public String toString() {
+        return "Request{" +
+                "method='" + method + '\'' +
+                ", endpoint='" + endpoint + '\'' +
+                '}';
+    }
+
+    static Request ping() {
+        return new Request("HEAD", "/", Collections.emptyMap(), null);
+    }
+
+    static Request get(GetRequest getRequest) {
+        StringJoiner pathJoiner = new StringJoiner("/", "/", "");
+        String endpoint = pathJoiner.add(getRequest.index()).add(getRequest.type()).add(getRequest.id()).toString();
+        Map<String, String> params = new HashMap<>();
+        params.put("ignore", "404");
+        putParam("preference", getRequest.preference(), params);
```

Comment:
All these parameters are later reused in the RequestTests, so I was wondering if if makes sense to extract them as constants? They are part of the rest api, so maybe can be public an get some javadoc?